### DISCUSSION TITLE
Roadmap for containerd image store out of experimental and enabled by default

### DISCUSSION THREAD

**Discussion by Romain-Geissler-1A** [moby/moby#48085] [Q&A]
Hi,

It seems one of the next "big thing" currently happening in the moby/docker development under the hood is the integration to containerd, in particular the move to the containerd image store. it seems some features at least on buildkit/buildx side seem to depend on this new image store if we decide to keep using the builtin builkit builder (which I find more user friendly than starting a container-base build server). I see for now containerd image store is still marked experimental and each new major releases comes with important fixes in that area, so I wouldn't enable it in "production enviroment (ie CI)" at scale for now. So this leads to my question, does Docker Inc or the moby community has some public plan to share about containerd image store about it's eventual move to production ready status and it's default enablement in the docker daemon.

Thanks !
Romain

**Comment by cpuguy83** [comment]
The main things that need to be addressed are related to polish.
Docker API's come from a time before multi-arch images which need to be retrofitted or replaced to take full advantage of containerd.

There's also some performance issues related to specific API's (like image listing) that will likely wind up informing new containerd API's, but I doubt would be problematic for most people.
